src/AutoRest.CSharp/Mgmt/AutoRest/MgmtOutputLibrary.cs

```diff
@@ -525,6 +598,73 @@ public bool TryGetRestClients(string requestPath, [MaybeNullWhen(false)] out Has
             return _requestPathToResources;
         }
 
+        private string? GetDefaultNameFromConfiguration(OperationSet operationSet, ResourceTypeSegment resourceType)
+        {
+            if (_context.Configuration.MgmtConfiguration.RequestPathToResourceName.TryGetValue(operationSet.RequestPath, out var name))
+                return name;
+            if (_context.Configuration.MgmtConfiguration.RequestPathToResourceName.TryGetValue($"{operationSet.RequestPath}|{resourceType}", out name))
+                return name;
+
+            return null;
+        }
+
+        private string EnsureResourceDefaultName(string candidateName, OperationSet operationSet, RequestPath requestPath)
+        {
+            // read configuration to see if we could get a configuration for this resource
+            var resourceType = requestPath.GetResourceType(_mgmtConfiguration);
+            var defaultNameFromConfig = GetDefaultNameFromConfiguration(operationSet, resourceType);
+            if (defaultNameFromConfig != null)
+                return defaultNameFromConfig;
+
+            var resourcesWithSameName = _resourceDataSchemaNameToOperationSets[candidateName];
+            var resourcesWithSameType = OperationSets.SelectMany(opSet =>
+            {
+                var operation = opSet.Operations.First();
+                return GetRequestPath(opSet[operation], operation).Expand(_mgmtConfiguration);
+            }).Where(rqPath =>
+            {
+                return rqPath.Count % 2 == 0 && rqPath.GetResourceType(_mgmtConfiguration).Equals(resourceType);
+            });
```

Comment:
I have some concerns about this assert of "request path must have even segments" - it is not true for a path of a scope resource, like `/{scope}/providers/Microsoft.Resources/policyAssignments/{name}` has odd segments, but it is a path of a resource.
How about we loop over the operation sets that are guaranteed to be a resource's?  Like this
```suggestion
            var resourcesWithSameType = _resourceDataSchemaNameToOperationSets.Values.SelectMany(opSet =>
            {
                var operation = opSet.Operations.First();
                return GetRequestPath(opSet[operation], operation).Expand(_mgmtConfiguration);
            }).Where(rqPath =>
            {
                return rqPath.Count % 2 == 0 && rqPath.GetResourceType(_mgmtConfiguration).Equals(resourceType);
            });
```